core/templates/services/html-length.service.ts

```diff
@@ -71,70 +78,67 @@ export class HtmlLengthService {
     const sanitizedHtml = this.sanitizer.sanitize(
       SecurityContext.HTML, htmlString) as string;
     let totalCharacters =
-    this.calculateBaselineLength(sanitizedHtml, 'character');
+    this.calculateBaselineLength(sanitizedHtml, CALCULATION_TYPE_CHARACTER);
     // Identify custom tags using regex on the original HTML string.
-    const customTagRegex = /<oppia-noninteractive-(?:math|image)[^>]*>/g;
-    const customTags = htmlString.match(customTagRegex);
+    const customTags = htmlString.match(CUSTOM_TAG_REGEX);
 
     if (customTags) {
       for (const customTag of customTags) {
         totalCharacters += this.getWeightForNonTextNodes(
-          customTag, 'character');
+          customTag, CALCULATION_TYPE_CHARACTER);
       }
     }
     return totalCharacters;
   }
 
   private calculateBaselineLength(
-      sanitizedHtml: string, wordOrCharacter: string): number {
+      sanitizedHtml: string, calculationType: CalculationType): number {
     let domparser = new DOMParser();
     let dom = domparser.parseFromString(sanitizedHtml, 'text/html');
     let totalWeight = 0;
     for (let tag of Array.from(dom.body.children)) {
       const ltag = tag.tagName.toLowerCase();
       if (this.textTags.includes(ltag)) {
         totalWeight +=
-        this.getWeightForTextNodes(tag as HTMLElement, wordOrCharacter);
+        this.getWeightForTextNodes(tag as HTMLElement, calculationType);
       }
     }
     return totalWeight;
   }
 
   private getWeightForTextNodes(
-      textNode: HTMLElement, wordOrCharacter: string): number {
+      textNode: HTMLElement, calculationType: CalculationType): number {
     const textContent = textNode.textContent || '';
 
-    if (wordOrCharacter === 'word') {
-      const words = textContent.trim().split(' ');
-      const wordCount = words.length;
-      return wordCount;
+    let textContentTrimmed = textContent.trim();
+    let textContentCount: number = 0;
+    if (calculationType === CALCULATION_TYPE_WORD) {
+      const words = textContentTrimmed.split(' ');
+      textContentCount = words.length;
+    } else if (calculationType === CALCULATION_TYPE_CHARACTER) {
+      const characters = textContentTrimmed.split('\n').join('').split('');
+      textContentCount = characters.length;
     }
-    const characters = textContent.trim().split('\n').join('').split('');
-    let characterCount = characters.length;
-    return characterCount;
+    return textContentCount;
```

Comment:
When will this be executed? There should probably be an "else throw Error" or similar if this case can't be reached; do not return an incorrect value.

Reply:
You still have to handle the error case here.

Reply:
done!

Reply:
Going back to the original question -- I still don't understand when this line will be executed. Can you clarify? Why would this be executed when there's an image in the string?

Also if there are only two possible calculation types then you should do if/else rather than if/elseif, and in the second else just assert that the calculation type is whatever it's supposed to be.

Reply:
> Please can you provide some more insight on when there's an image in the string? as we are already separating the text nodes and non-text nodes before passing anything into the function.

You wrote "This will be executed when the hint has an image" in an earlier comment, and I don't understand what the image has to do with it.

-----

> I have used elseif block thinking that if in future we require another counting type like sentence or any other. If you want I will change that to else block.

Use else and an assertion, as requested previously. If we need a new type we can always add a new elseif clause, but we need a catchall at the end to ensure all cases are covered.